internalshared/configutil/listener.go

```diff
@@ -315,6 +315,18 @@ func ParseListeners(result *SharedConfig, list *ast.ObjectList) error {
 					return multierror.Prefix(fmt.Errorf("error parsing x_forwarded_for_authorized_addrs: %w", err), fmt.Sprintf("listeners.%d", i))
 				}
 
+				for _, m := range l.XForwardedForAuthorizedAddrs {
```

Comment:
I'm not sure I'm following you. How would we marshal and validate in one shot? Are you saying we should change `parseutil.ParseAddrs` instead of adding this bit here? If so, my main hesitation with doing that is that it's a shared library, used by Vault, Boundary, and possibly other things as well, so there might be other things relying on its current behavior. But it's possible I just haven't understood your comment.

Reply:
Well, that was my initial thought, but yeah it's external and probably depended upon by a bunch of other stuff.

I guess we could add a new `ParseTypedAddrs([]*sockaddr.SockAddrMarshaler, sat SockAddrType)` and call it where appropriate. BUT:
1. I wasn't sure if there was some other magic you could think of.
2. It probably violates YAGNI